spack/packages/dla-future/package.py

```diff
@@ -16,23 +15,37 @@ class DlaFuture(CMakePackage):
 
     version('develop', branch='master')
 
-    variant('gpu', default=False,
+    variant('cuda', default=False,
             description='Use the GPU/cuBLAS back end.')
+    variant('doc', default=False,
+            description='Build documentation.')
 
-    # Until mpich is default comment this out
     #depends_on('mpi@3:')
-    depends_on('mpich')
-    depends_on('mkl')
+    depends_on('mpi')
     depends_on('blaspp')
     depends_on('lapackpp')
-    depends_on('hpx cxxstd=14 networking=none')
-    depends_on('cuda', when='gpu=True')
+    depends_on('hpx@1.4.0:1.4.1 max_cpu_count=128 cxxstd=14 networking=none')
```

Comment:
I think `max_cpu_count` constraint is too strict.

This would not allow to depend on an hpx installation that has been compiled with any other value for this variant. 

`spack spec dla-future ^ hpx@1.4.1 max_cpu_count=256`

Let's say that on my particular platform I need to build hpx with `max_cpu_count=256`, this constraint would not allow me to do it.

Reply:
Right.